multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -108,6 +111,8 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 	r.mapLock.Lock()
 	defer r.mapLock.Unlock()
 
+	// If !ok, it means member not found. If this happens, the MemberClusterAnnounce should soon be deleted.
+	// Nothing to do here.
 	if data, ok := r.timerData[common.ClusterID(memberAnnounce.ClusterID)]; ok {
```

Comment:
@luolanzone : I feel we should revisit all these cluster status code. I have the following question:

1. Do we really need leader cluster ID in the MemberAnnounce?
2. Do we still need all status conditions after we remove election?
3. Add/RemoveMember() need not to be called by ClusterSet changes, after we change to auto-update ClusterSet, but we can start to maintain the timerdata once MemberClusterAnnounce is created?

Could you please look into the code to understand the whole status update logic, and clean up the code?

Reply:
Maybe you can first summarize what conditions we have, and they are set for what cases.

Reply:
@luolanzone : a reminder for you.